plasmapy/utils/checks.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+from astropy import units as u
+from ..constants import c
+
+
+def _check_quantity(arg, argname, funcname, units, can_be_negative=True,
+                    can_be_complex=False, can_be_inf=True):
+    """Raises exceptions if an object is not an astropy Quantity with
+    correct units and valid numerical values.
+
+    Parameters
+    ----------
+    arg : Quantity
+        The object to be tested.
+
+    argname : string
+        The name of the argument to be printed in error messages.
+
+    funcname : string
+        The name of the original function to be printed in error messages.
+
+    units : Unit or list of Units
+        Acceptable units for arg.
+
+    can_be_negative : boolean, optional
+        True if the Quantity can be negative, False otherwise.
+        Defaults to True.
+
+    can_be_complex : boolean, optional
+        True if the Quantity can be a complex number, False otherwise.
+        Defaults to False.
+
+    can_be_inf : boolean, optional
+        True if the Quantity can contain infinite values, False
+        otherwise.  Defaults to True.
+
+    Raises
+    ------
+    TypeError
+        If the argument is not a Quantity or units is not entirely units.
+
+    UnitConversionError
+        If the argument is not in acceptable units.
+
+    ValueError
+        If the argument contains NaNs or other invalid values as
+        determined by the keywords.
+
```

Comment:
A nice touch that this doesn't simply return `True` but does all the testing inside.